### PR TITLE
909 - Added db migration for bcrhp to change all instances of /int/bcrhp/ to /bcrhp/

### PR DESCRIPTION
**Pull request by JessicaInTime**
bcgov/BCHeritage#909

This includes updating tables: map_layers, map_sources, cards_x_nodes_x_widgets, tiles, published_graphs, nodes, values Updating V_HISTORIC_SITE site_url column

Did not update the edit_log table.

### CODE REVIEW

File: src/bcrhp/bcrhp/migrations/909_update_url_prefix.py [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+    dependencies = [
+        ('bcrhp', '947_historic_site_one_line'),
+    ]
+
+    operations = [
+        RunPrivilegedSQL(
+            """
+            begin;
+            update map_layers
+            set legend = replace(legend, '/int/bcrhp/', '/bcrhp/');
```
Comment (bferguso) [inline src/bcrhp/bcrhp/migrations/909_update_url_prefix.py]: @jrods - please restrict the update clause to all updates to limit the updates to the rows that will be affected. Eg add a clause like `where legend ~ '/int/bcrhp/` to the updates.

File: src/bcrhp/bcrhp/migrations/909_update_url_prefix.py [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+            begin;
+            update map_layers
+            set legend = replace(legend, '/bcrhp/', '/int/bcrhp/');
+            commit;
+            """
+        ),
+        RunPrivilegedSQL(
+            """
+            begin;
+            update map_sources
```
Comment (bferguso) [inline src/bcrhp/bcrhp/migrations/909_update_url_prefix.py:34]: This query nulls out map sources: mapbox-satellite, search-query and geocode-point values. Needs to be limited to rows it should change.

File: src/bcrhp/bcrhp/migrations/909_update_url_prefix.py [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+            update concepts
+            set legacyoid = replace(legacyoid, '/bcrhp/', '/int/bcrhp/')
+            ;
+            commit;
+            """
+        ),
+        RunPrivilegedSQL(
+            """
+            begin;
+            update values
```
Comment (bferguso) [inline src/bcrhp/bcrhp/migrations/909_update_url_prefix.py:263]: As above - please limit all updates to rows that should be updated.

File: src/bcrhp/bcrhp/migrations/909_update_url_prefix.py [hunk 4]
```diff
@@ -0,0 +1,542 @@
...
+            commit;
+            """,
+            """
+            begin;
+            update map_layers
+            set legend = replace(legend, '/bcrhp/', '/int/bcrhp/');
+            commit;
+            """
+        ),
+        RunPrivilegedSQL(
```
Comment (bferguso) [inline src/bcrhp/bcrhp/migrations/909_update_url_prefix.py]: The RunPrivilegedSQL is only required when making DDL changes. Data updates should be done with RunSQL as the default user has access to modify data.

**Review by bferguso — CHANGES_REQUESTED** [review]
Looking pretty good!

Please update the graph json files in pkg/graphs to keep them current with what is in the database and add a line to the readme. 

A few of the map sources values were nulled out which bricked my system. Please ensure that the updates for the remaining tables don't have any unexpected side effects.

File: src/bcrhp/bcrhp/migrations/909_update_url_prefix.py (hunk 1 above)
Comment (JessicaInTime) [reply src/bcrhp/bcrhp/migrations/909_update_url_prefix.py]: Will do

File: src/bcrhp/bcrhp/migrations/909_update_url_prefix.py (hunk 3 above)
Comment (JessicaInTime) [reply src/bcrhp/bcrhp/migrations/909_update_url_prefix.py:263]: Done

File: src/bcrhp/bcrhp/migrations/909_update_url_prefix.py (hunk 4 above)
Comment (JessicaInTime) [reply src/bcrhp/bcrhp/migrations/909_update_url_prefix.py]: Updated all the data updates and just left the v_historic_site using the priv ops

File: src/bcrhp/bcrhp/migrations/909_update_url_prefix.py (hunk 2 above)
Comment (JessicaInTime) [reply src/bcrhp/bcrhp/migrations/909_update_url_prefix.py:34]: Updated

**Review by bferguso — APPROVED** [review]
Looks great!

**Pull request merged**